core/state_processor_rip7560.go

```diff
@@ -471,10 +484,16 @@ func validatePaymasterReturnData(data []byte) (context []byte, validAfter, valid
 		return nil, 0, 0, errors.New("invalid paymaster return data length")
 	}
 	validationData, context := UnpackPaymasterValidationReturn(data)
-	magicExpected, validAfter, validUntil := UnpackValidationData(validationData)
+	if validationData == nil {
+		return nil, 0, 0, errors.New("invalid paymaster return data")
+	}
+	magicExpected, validUntil, validAfter := UnpackValidationData(validationData)
 	if magicExpected != MAGIC_VALUE_PAYMASTER {
 		return nil, 0, 0, errors.New("paymaster did not return correct MAGIC_VALUE")
 	}
+	if len(context) > PAYMASTER_MAX_CONTEXT_SIZE {
+		return nil, 0, 0, errors.New("paymaster context too large")
+	}
```

Comment:
I repeat myself but I think some of these checks should be done inside the `UnpackPaymasterValidationReturn` function, no need to return invalid values only to error out later.